server/src/main/java/com/objectcomputing/checkins/services/action_item/ActionItemServicesImpl.java

```diff
@@ -82,14 +124,70 @@ public ActionItem update(ActionItem actionItem) {
     }
 
     public Set<ActionItem> findByFields(UUID checkinid, UUID createdbyid) {
+        String workEmail = securityService != null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail != null ? currentUserServices.findOrSaveUser(null, workEmail) : null;
+        Boolean isAdmin = securityService != null && securityService.hasRole(RoleType.Constants.ADMIN_ROLE);
+
+        if (checkinid != null) {
+            CheckIn checkinRecord = checkinRepo.findById(checkinid).orElse(null);
+            final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+            final UUID teamMemberId = checkinRecord != null ? checkinRecord.getTeamMemberId() : null;
+            validate(!currentUser.getId().equals(pdlId) &&
+                    !currentUser.getId().equals(teamMemberId) &&
+                    !isAdmin, "User is unauthorized to do this operation");
+        } else if (createdbyid != null) {
+            MemberProfile memberRecord = memberRepo.findById(createdbyid).orElse(null);
+            validate(!currentUser.getId().equals(memberRecord.getId()) &&
+                    !isAdmin, "User is unauthorized to do this operation");
+        } else {
+            validate(!isAdmin, "User is unauthorized to do this operation");
+        }
 
-        return new LinkedHashSet<>(
+        Set<ActionItem> actionItems = new LinkedHashSet<>(
                 actionItemRepo.search(nullSafeUUIDToString(checkinid), nullSafeUUIDToString(createdbyid)));
+
+        return actionItems;
+
     }
 
     public void delete(@NotNull UUID id) {
+
+        String workEmail = securityService != null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail != null ? currentUserServices.findOrSaveUser(null, workEmail) : null;
+        Boolean isAdmin = securityService != null && securityService.hasRole(RoleType.Constants.ADMIN_ROLE);
+        ActionItem actionItem = actionItemRepo.findById(id).orElse(null);
+
+        validate(actionItem == null, "invalid action item %s", id);
+
+        final UUID checkinId = actionItem.getCheckinid();
+        final UUID createById = actionItem.getCreatedbyid();
+
+        CheckIn checkinRecord = checkinRepo.findById(checkinId).orElse(null);
+        Boolean isCompleted = checkinRecord != null ? checkinRecord.isCompleted() : null;
+        final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+        final UUID teamMemberId = checkinRecord != null ? checkinRecord.getTeamMemberId() : null;
```

Comment:
variable teamMemberId could be deleted it is not used

Reply:
good catch - updated